bindings/go/tests/listener_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2021, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+package goucxtests
+
+import (
+	"net"
+	"testing"
+	. "ucx"
+)
+
+var connReq *UcpConnectionRequest = nil
+var rejected bool = false
+
+func TestUcpListener(t *testing.T) {
+	addr, _ := net.ResolveTCPAddr("tcp", "0.0.0.0")
+
+	ucpParams := (&UcpParams{}).EnableTag()
+
+	ucpWorkerParams := (&UcpWorkerParams{}).SetThreadMode(UCS_THREAD_MODE_SINGLE)
+
+	listenerParams := &UcpListenerParams{}
+	listenerParams.SetSocketAddress(addr)
+
+	epParams := &UcpEpParams{}
+	var logErrorHandler UcpEpErrHandler = func(ep *UcpEp, status UcsStatus) {
+		t.Logf("Error handler called with status %d", status)
+	}
+	epParams.SetPeerErrorHandling().SetErrorHandler(logErrorHandler)
+
+	listenerParams.SetConnectionHandler(func(connRequest *UcpConnectionRequest) {
+		// Use first request to create backward ep with epParams.SetConnRequest()
+		// And second request reject to test ep error handling.
+		if connReq == nil {
+			connReq = connRequest
+		} else {
+			connRequest.Reject()
+		}
+	})
+
+	context1, _ := NewUcpContext(ucpParams)
+	context2, _ := NewUcpContext(ucpParams)
+	defer context1.Close()
+	defer context2.Close()
+
+	worker1, _ := context1.NewWorker(ucpWorkerParams)
+	worker2, _ := context2.NewWorker(ucpWorkerParams)
+	defer worker1.Close()
+	defer worker2.Close()
+
+	listener, _ := worker1.NewListener(listenerParams)
+	defer listener.Close()
+
+	listenerAttrs, _ := listener.Query(UCP_LISTENER_ATTR_FIELD_SOCKADDR)
+
+	listenerAddress := listenerAttrs.Address
+
+	t.Logf("Started listener on address: %v", listenerAddress)
+
+	epParams.SetSocketAddress(listenerAddress)
+
+	ep, err := worker2.NewEndpoint(epParams)
+	if err != nil {
+		t.Fatalf("Can't create endpoint %v", err)
+	}
+
+	flushReq1, _ := ep.FlushNonBlocking(nil)
+	defer flushReq1.Close()
+	for connReq == nil {
+		worker1.Progress()
+		worker2.Progress()
+	}
+
+	backwardEpParams := (&UcpEpParams{}).SetConnRequest(connReq)
+	backwardEpParams.SetPeerErrorHandling().SetErrorHandler(logErrorHandler)
```

Comment:
"backwardEp" means "peerEp"/"replyEp"?

Reply:
It's ep created from connectionRequest on a server.